.github/workflows/contracts-e2e-tests-and-deploy.yaml

```diff
@@ -294,9 +309,14 @@ jobs:
           -H 'Authorization: Bearer ${{ secrets.CI_GH_TOKEN }}' \
           -d '{ "ref":"master" }'
 
-      - name: Trigger The Button deployment workflow
+      - name: Setup kubectl
+        uses: azure/setup-kubectl@v2.0
+        with:
+          version: 'v1.23.6'
+
+      - name: Restart the-button deployment
+        shell: bash
         run: |
-          curl -X POST 'https://api.github.com/repos/Cardinal-Cryptography/the-button/actions/workflows/main.yaml/dispatches' \
-          -H "Accept: application/vnd.github+json" \
-          -H 'Authorization: Bearer ${{ secrets.CI_GH_TOKEN }}' \
-          -d '{ "ref":"main", "inputs": { "buildImage": "false", "deployImage": "true" }}'
+          aws eks --region eu-central-1 update-kubeconfig --name alephzero-devnet-eu-central-1-eks
```

Comment:
Will this still build a new image, with say updated metadata and/or addresses ?

k8 is over my head, best if someone else takes a look too (every PR should have two reviewers as a general-ish rule anyway)

Reply:
No, it will not build new the-button docker image. JSON files are now mounted into the the-button container. Pod in kubernetes that runs the-button container, has got an additional init container which executes before running the-button, which downloads the JSON files from S3 bucket. This whole thing sits in `aleph-apps` repository (and the init container dockerfile, simirarly to other init containers, is in `aleph-infra`).
For example, we have JSON here https://the-button.dev.azero.dev/metadata/addresses.json (same container, no additional endpoints). For `featurenet` environment, the file is downloaded with `fetch`  at the start of the-button frontend.
With that, the-button workflow is free of any featurenet steps (like downloading JSON files from S3) and the-button docker image does not have to be rebuilt. The running application (its pod) just have to be restarted.

Reply:
Also, the code here triggers the re-deployment of the-button straight away. And waits for its success in the next line.

Reply:
@Kosiare application does not have to be updated. just the init container needs to be restarted.